<a href="https://colab.research.google.com/github/kevin4965/INGWEBII/blob/master/analisiscovid19.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

JOSE QUEVIN ORDOÑEZ SANCHEZ
PREICA2501B010058
Análisis exploratorio de datos del Covid-19 en Colombia
Analítica de Datos Para Ingeniería Software y Datos
Andrés Felipe Palacio
Institución Universitaria Digital de Antioquia
Medellín, Antioquia.
2025






**EJERCICIO 1**

**Importa las siguientes librerías: Numpy, Pandas, MatplotLib.**

In [2]:
#importar librerias
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

**Carga los datos desde el sitio web que te sugerimos.**

In [3]:
datos = pd.read_csv('https://www.datos.gov.co/api/views/gt2j-8ykr/rows.csv?accessType=DOWNLOAD&bom=true&format=true')

<ipython-input-3-f02b5dc7a296>:1: DtypeWarning: Columns (13,14) have mixed types. Specify dtype option on import or set low_memory=False.
  datos = pd.read_csv('https://www.datos.gov.co/api/views/gt2j-8ykr/rows.csv?accessType=DOWNLOAD&bom=true&format=true')


**Obtén la información estructural del DataFrame con el método .info(), y describe los resultados**

In [5]:
print (datos.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6390971 entries, 0 to 6390970
Data columns (total 23 columns):
 #   Column                        Dtype  
---  ------                        -----  
 0   fecha reporte web             object 
 1   ID de caso                    object 
 2   Fecha de notificación         object 
 3   Código DIVIPOLA departamento  object 
 4   Nombre departamento           object 
 5   Código DIVIPOLA municipio     object 
 6   Nombre municipio              object 
 7   Edad                          int64  
 8   Unidad de medida de edad      int64  
 9   Sexo                          object 
 10  Tipo de contagio              object 
 11  Ubicación del caso            object 
 12  Estado                        object 
 13  Código ISO del país           object 
 14  Nombre del país               object 
 15  Recuperado                    object 
 16  Fecha de inicio de síntomas   object 
 17  Fecha de muerte               object 
 18  Fecha de diagnóstico  

El método .info() en Pandas es una herramienta muy útil para obtener un resumen rápido de un DataFrame. Proporciona información sobre la estructura del DataFrame, incluyendo el número de filas, el número de columnas, los nombres de las columnas, los tipos de datos de cada columna, la cantidad de valores no nulos y el uso de memoria.
En resumen vemos que el objeto es un Dataframe de Pandas, tiene 6,390,971 filas, con índices que van desde 0 hasta 6,390,970, tiene 23 columnas, vemos que hay 3 tipos de datos object, int64, float64, El DataFrame está utilizando aproximadamente 1.1 gigabytes de memoria.
Las columnas de tipo object contienen principalmente texto o fechas, que Pandas interpreta como objetos.

**Visualiza la estructura del DataFrame (.head(),.tail() y .sample())**



In [6]:
# Mostrar las primeras 5 filas del DataFrame
print(datos.head())

     fecha reporte web ID de caso Fecha de notificación  \
0  2020-12-24 00:00:00  1,556,979   2020-12-22 00:00:00   
1  2020-12-24 00:00:00  1,556,980   2020-12-19 00:00:00   
2  2020-12-24 00:00:00  1,556,981   2020-12-19 00:00:00   
3  2020-12-24 00:00:00  1,556,982   2020-12-22 00:00:00   
4  2020-12-24 00:00:00  1,556,983   2020-12-22 00:00:00   

  Código DIVIPOLA departamento Nombre departamento Código DIVIPOLA municipio  \
0                           76               VALLE                    76,001   
1                           76               VALLE                    76,001   
2                           76               VALLE                    76,001   
3                           76               VALLE                    76,001   
4                           76               VALLE                    76,001   

  Nombre municipio  Edad  Unidad de medida de edad Sexo  ...  \
0             CALI    67                         1    F  ...   
1             CALI    66            

.head()
Muestra las primeras filas del DataFrame (por defecto, las primeras 5 filas).
Nos permite ver cómo comienzan los datos.
Útil para verificar si los datos se cargaron correctamente y para entender la estructura inicial.

In [7]:
# Mostrar las últimas 5 filas del DataFrame
print(datos.tail())

           fecha reporte web ID de caso Fecha de notificación  \
6390966  2021-10-24 00:00:00  4,990,088   2021-10-19 00:00:00   
6390967  2021-10-24 00:00:00  4,990,091   2021-10-23 00:00:00   
6390968  2021-10-24 00:00:00  4,990,092   2021-10-19 00:00:00   
6390969  2021-10-24 00:00:00  4,990,095   2021-11-05 00:00:00   
6390970  2021-10-24 00:00:00  4,990,096   2021-10-22 00:00:00   

        Código DIVIPOLA departamento Nombre departamento  \
6390966                           76               VALLE   
6390967                           76               VALLE   
6390968                           76               VALLE   
6390969                           76               VALLE   
6390970                           76               VALLE   

        Código DIVIPOLA municipio Nombre municipio  Edad  \
6390966                    76,520          PALMIRA    37   
6390967                    76,520          PALMIRA    36   
6390968                    76,364          JAMUNDI    39   
6390969 

.tail()
Muestra las últimas filas del DataFrame (por defecto, las últimas 5 filas).
Nos permite ver cómo terminan los datos.
Útil para verificar si hay algún problema con los datos al final del archivo.

In [8]:
# Mostrar una fila aleatoria del DataFrame
print(datos.sample())

           fecha reporte web ID de caso Fecha de notificación  \
1181686  2021-05-15 00:00:00  3,102,940   2021-05-02 00:00:00   

        Código DIVIPOLA departamento Nombre departamento  \
1181686                           17              CALDAS   

        Código DIVIPOLA municipio Nombre municipio  Edad  \
1181686                    17,001        MANIZALES    35   

         Unidad de medida de edad Sexo  ... Código ISO del país  \
1181686                         1    F  ...                 NaN   

        Nombre del país  Recuperado Fecha de inicio de síntomas  \
1181686             NaN  Recuperado         2021-04-28 00:00:00   

        Fecha de muerte Fecha de diagnóstico Fecha de recuperación  \
1181686             NaN  2021-05-13 00:00:00   2021-05-16 00:00:00   

        Tipo de recuperación Pertenencia étnica Nombre del grupo étnico  
1181686               Tiempo                6.0                     NaN  

[1 rows x 23 columns]


.sample()
Muestra una fila aleatoria del DataFrame.
Nos da una vista aleatoria de los datos.
Útil para inspeccionar filas en diferentes partes del DataFrame sin sesgo.

**Utiliza value_counts(dropna=False), para visualizar las siguientes variables: Estado, Atención, Sexo, Tipo, País_procedencia, Tipo_recuperación, Pertenencia_étnica, Nombre_grupo_étnico**

In [10]:
# Lista de columnas a analizar
columns_to_analyze = [
    'Estado', 'Sexo', 'Tipo de contagio', 'Nombre del país',
    'Tipo de recuperación', 'Pertenencia étnica', 'Nombre del grupo étnico'
]

# Iterar sobre cada columna y aplicar value_counts(dropna=False)
for column in columns_to_analyze:
    print(f"Distribución de valores en la columna '{column}':")
    print(datos[column].value_counts(dropna=False))
    print("\n" + "="*50 + "\n")  # Separador visual

Distribución de valores en la columna 'Estado':
Estado
Leve         6154791
Fallecido     143125
leve           50623
NaN            41268
Moderado        1056
Grave            106
LEVE               2
Name: count, dtype: int64


Distribución de valores en la columna 'Sexo':
Sexo
F    3417087
M    2973872
f          6
m          6
Name: count, dtype: int64


Distribución de valores en la columna 'Tipo de contagio':
Tipo de contagio
Comunitaria    4447740
Relacionado    1939528
Importado         3703
Name: count, dtype: int64


Distribución de valores en la columna 'Nombre del país':
Nombre del país
NaN                          6387253
ESTADOS UNIDOS DE AMÉRICA       1418
MÉXICO                           385
ESPAÑA                           370
BRASIL                           285
                              ...   
ANDORRA                            1
KENYA                              1
GUADALUPE                          1
REPÚBLICA DOCIMINCANA              1
HAITÍ                   

el método .value_counts(dropna=False) cuenta la frecuencia de cada valor único en una columna, incluyendo los valores NaN (valores faltantes) si dropna=False.

value_counts() cuenta cuántas veces aparece cada valor en la columna.
dropna=False asegura que también se cuenten los valores nulos (NaN).

Este método es muy útil para analizar datos categóricos y detectar valores nulos dentro de un conjunto de datos.